TestProcessMetadataManager.test_deadline_until is flaky

Failed on #6800: https://travis-ci.org/pantsbuild/pants/jobs/461128648. A solved, probably unrelated error in this test can be seen at #5417.

Can this test expand its upper and lower bounds here? Since the error is about receiving 3 instead of 4-6 messages, this seems like prime flakiness material.
```
==================== FAILURES ====================
_ TestProcessMetadataManager.test_deadline_until _

self = <pants_test.pantsd.test_process_manager.TestProcessMetadataManager testMethod=test_deadline_until>

    def test_deadline_until(self):
      with self.assertRaises(ProcessMetadataManager.Timeout):
        with self.captured_logging(logging.INFO) as captured:
          self.pmm._deadline_until(lambda: False, 'the impossible', timeout=.5, info_interval=.1)
      self.assertTrue(4 <= len(captured.infos()) <= 6,
>                     'Expected between 4 and 6 infos, got: {}'.format(captured.infos()))
E     AssertionError: False is not true : Expected between 4 and 6 infos, got: ['pants.pantsd.process_manager: waiting for the impossible...', 'pants.pantsd.process_manager: waiting for the impossible...', 'pants.pantsd.process_manager: waiting for the impossible...']

pants_test/pantsd/test_process_manager.py:147: AssertionError
--------------- Captured log call ----------------
process_manager.py         138 INFO     waiting for the impossible...
process_manager.py         138 INFO     waiting for the impossible...
process_manager.py         138 INFO     waiting for the impossible...
 generated xml file: /home/travis/build/pantsbuild/pants/.pants.d/test/pytest/tests.python.pants_test.pantsd.process_manager/junitxml/TEST-tests.python.pants_test.pantsd.process_manager.xml 
```

Any approach to more robustly test this sort of ipc seems like it could be dealt with at least partially in a similar way to any code that e.g. retries network requests (and there's no reason this code would need to only be usable in tests).

--- a/tests/python/pants_test/pantsd/test_process_manager.py
+++ b/tests/python/pants_test/pantsd/test_process_manager.py
@@ -1,5 +1,6 @@
 # Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
+
 import errno
 import logging
 import os
@@ -28,13 +29,24 @@ def fake_process(**kwargs):
     return proc
 
 
+# For history of this test's flakiness: https://github.com/pantsbuild/pants/issues/6836
+def test_deadline_until(caplog):
+    pm = ProcessManager("_test_", metadata_base_dir=safe_mkdtemp())
+    with pytest.raises(ProcessManager.Timeout):
+        with caplog.at_level(logging.INFO):
+            pm._deadline_until(
+                lambda: False, "the impossible", "completed? how?", timeout=0.5, info_interval=0.1
+            )
+    assert 4 <= len(caplog.records) <= 6, f"Expected between 4 and 6 infos, got: {caplog.records}"
+
+
+# TODO: Convert the rest of these tests to pytest-style test_* functions.
 class TestProcessManager(unittest.TestCase):
     NAME = "_test_"
     TEST_KEY = "TEST"
     TEST_VALUE = "300"
     TEST_VALUE_INT = 300
     BUILDROOT = "/mock_buildroot/"
-    SUBPROCESS_DIR = safe_mkdtemp()
 
     def setUp(self):
         super().setUp()
@@ -62,19 +74,6 @@ class TestProcessManager(unittest.TestCase):
             self.assertEqual(
                 self.pmm.read_metadata_by_name(self.TEST_KEY, int), self.TEST_VALUE_INT
             )
-
-    @pytest.mark.skip(reason="flaky: https://github.com/pantsbuild/pants/issues/6836")
-    @pytest.mark.no_error_if_skipped
-    def test_deadline_until(self):
-        with self.assertRaises(ProcessManager.Timeout):
-            with self.captured_logging(logging.INFO) as captured:
-                self.pmm._deadline_until(
-                    lambda: False, "the impossible", timeout=0.5, info_interval=0.1
-                )
-        self.assertTrue(
-            4 <= len(captured.infos()) <= 6,
-            f"Expected between 4 and 6 infos, got: {captured.infos()}",
-        )
 
     def test_wait_for_file(self):
         with temporary_dir() as td:

